### PR TITLE
feat(#2506): copy sources mojo

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Ref: #2506 

What's done:
- Introduced `CopySourcesMojo` that copies sources to `target/generated-sources` folder and adds extra folder on top
- Added integration test for testing whole pipeline described in the ticket

---

## PR-Codex overview
This PR focuses on adding a new package layer to all source files. 

### Detailed summary
- Added a new package layer to all source files.
- Updated invoker goals in the properties file.
- Added a logo image and test description in the README file.
- Updated CommitHashesMap.java with a new commit hash.
- Updated the exclude_paths in the codacy.yml file.
- Added package-info.java files to the source files.
- Added annotations to certain Java files.
- Added verify.groovy file for verification.
- Added new Java files for attribute and phi classes.

> The following files were skipped due to too many changes: `eo-maven-plugin/src/it/hash_package_layer/src/main/java/org/eolang/Phi.java`, `eo-maven-plugin/src/it/hash_package_layer/pom.xml`, `eo-maven-plugin/src/it/hash_package_layer/src/main/java/EOorg/EOeolang/Heaps.java`, `eo-maven-plugin/src/main/java/org/eolang/maven/CopiedResources.java`, `eo-maven-plugin/src/main/java/org/eolang/maven/CopySourcesMojo.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please review this one

File: eo-maven-plugin/src/main/java/org/eolang/maven/CopiedResources.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                            "resource",
+                            MojoExecutor.element("directory", sources.toString()),
+                            MojoExecutor.element("filtering", "true")
+                        )
+                    )
+                ),
+                this.environment
+            );
+        } catch (final MojoExecutionException ex) {
+            throw new IllegalStateException(ex);
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/CopiedResources.java:86]: @maxonfjvipon Let's add a meaningful message here. According to: https://www.yegor256.com/2015/12/01/rethrow-exceptions.html#:~:text=This%20is%20how%20they,should%20look%20like%20this%3A

**Review by Graur — CHANGES_REQUESTED** [review]
@maxonfjvipon Good one! Just one comment from my side:

**Comment by maxonfjvipon** [comment]
@Graur please have a look one more time

**Review by Graur — APPROVED** [review]
@maxonfjvipon Looks good to me!

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**